### PR TITLE
feat: connect database and create models

### LINKED ISSUE #1: feat: connect db and create models
**Create models:**
- user
- course
- section
- subsection
- user_enrolled_courses
- user_bookmarked_courses
- course_votes
- user_progress
- course_types

### PR DESCRIPTION
**Pull request by siddhigate**
Created models:
- user
- course
- section
- sub_section
- user_enrolled_courses
- user_bookmarked_courses
- user_progress
- course_votes
- course_types

### CODE REVIEW

File: database/models/course.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      public_id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
```
Comment (dushyant0814) [inline database/models/course.js]: type and defaultValue datatype mismatch

File: database/config/config.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const {DB, DB_PASSWORD, DB_HOST, DB_PORT, DB_USER } = process.env;
+
+const config = {
```
Comment (dushyant0814) [inline database/config/config.js:3]: use connection pooling too, read about it.

File: database/models/course.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      creator_id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        references: {
+          model: "users",
+          key: "id",
+        },
+      },
+      types: {
+        type: DataTypes.ARRAY(DataTypes.UUID),
```
Comment (dushyant0814) [inline database/models/course.js:40]: filtering possible in arrays?

File: database/models/courseType.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      public_id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
```
Comment (dushyant0814) [inline database/models/courseType.js]: type and default value mismatch

File: database/models/section.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      public_id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
```
Comment (dushyant0814) [inline database/models/section.js]: mismatch type and default value

File: database/models/subsection.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    {
+      id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      public_id: {
+        type: DataTypes.UUID,
+        defaultValue: DataTypes.UUIDV4,
```
Comment (dushyant0814) [inline database/models/subsection.js]: same

**Review by dushyant0814 — CHANGES_REQUESTED** [review]

File: database/models/course.js (hunk 3 above)
Comment (siddhigate) [reply database/models/course.js:40]: Yes. Tried using Op.contains operator.

File: database/config/config.js (hunk 2 above)
Comment (siddhigate) [reply database/config/config.js:3]: Done and read about it👍

File: database/models/course.js (hunk 1 above)
Comment (siddhigate) [reply database/models/course.js]: Fixed

File: database/models/section.js (hunk 5 above)
Comment (siddhigate) [reply database/models/section.js]: Fixed

File: database/models/subsection.js (hunk 6 above)
Comment (siddhigate) [reply database/models/subsection.js]: Fixed

**Review by dushyant0814 — APPROVED** [review]

**Pull request merged**